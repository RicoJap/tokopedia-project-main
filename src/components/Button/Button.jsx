import React from "react";
import MuiButton from "@material-ui/core/Button";

/**
 * Button Component
 *
 * @since  14/11/2020
 * @access (public)
 *
 * @type     Functional Component
 * @param {String}  label  Label of the button
 * @param {String}  variant  variations of the button (text, outlined, filled)
 *
 */

const Button = ({ label, variant = "outlined", ...attributes }) => {
  return (
    <MuiButton variant={variant} color="primary" {...attributes}>
      {label}
    </MuiButton>
  );
};

export default Button;
