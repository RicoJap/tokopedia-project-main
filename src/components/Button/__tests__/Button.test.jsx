import "@testing-library/jest-dom";
import React from "react";
import renderer from "react-test-renderer";

import { fireEvent, render, screen, waitFor } from "@testing-library/react";

import Button from "../Button";

const mockOnClick = jest.fn();

const MockButton = () => {
  return <Button label={"Test Button"} onClick={mockOnClick} />;
};

describe("Snapshot testing", () => {
  const tree = renderer.create(<MockButton />).toJSON();
  expect(tree).toMatchSnapshot();
});

describe("Button Component renders correctly", () => {
  test("Check if Button component is present in the document", () => {
    render(<MockButton />);
    expect(
      screen.getByRole("button", { name: /test button/i })
    ).toBeInTheDocument();
  });
});

describe("Test Button component event handler", () => {
  test("Check if Button onClick is called when clicked", () => {
    render(<MockButton />);
    fireEvent.click(screen.getByRole("button", { name: /test button/i }));
    expect(mockOnClick).toHaveBeenCalledTimes(1);
  });
});
