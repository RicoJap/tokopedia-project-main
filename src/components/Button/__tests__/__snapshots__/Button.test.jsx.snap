// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
  data-testid="button"
  disabled={false}
  onBlur={[Function]}
  onClick={[MockFunction]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  >
    Test Button
  </span>
</button>
`;
